### ISSUE TITLE
Strange Molecular pose output

### ISSUE THREAD

**Issue by PhungVanDuy** [luwei0917/TankBind#19]

**Comment by spadavec** [comment]
![gtp](https://user-images.githubusercontent.com/3453650/213876416-20c70912-8fbd-4778-8f5d-ce3279096315.png)

I tried to dock GTP to a KRAS structure and got a very strange geometry for the base. 
@luwei0917 any ideas what might be causing this? 

@PhungVanDuy have you managed to figure this out?

**Comment by dhruv-acog** [comment]
I had a similar issue, this is what my output pose looks like:

![image](https://user-images.githubusercontent.com/122432159/229080194-446aa39e-bb36-44ea-89c0-e497783a4b19.png)
 
Chimera:
![image](https://user-images.githubusercontent.com/122432159/229080361-59ea5d3c-6a0c-433f-8f4a-719b16dfd316.png)

This is the code I used to generate my output
`
idx = 0
line = info.iloc[idx]
one_data = dataset[idx]
data_with_batch_info = next(iter(DataLoader(dataset[idx:idx+1], batch_size=1, 
                             follow_batch=['x', 'y', 'compound_pair'], shuffle=False, num_workers=1)))
y_pred, affinity_pred = model(data_with_batch_info)

coords = one_data.coords.to(device)
protein_nodes_xyz = one_data.node_xyz.to(device)
n_compound = coords.shape[0]
n_protein = protein_nodes_xyz.shape[0]
y_pred = y_pred.reshape(n_protein, n_compound).to(device).detach()
y = one_data.dis_map.reshape(n_protein, n_compound).to(device)
compound_pair_dis_constraint = torch.cdist(coords, coords)

smiles = line['smiles']
mol = Chem.MolFromSmiles(smiles)
mol.Compute2DCoords()
LAS_distance_constraint_mask = get_LAS_distance_constraint_mask(mol).bool()
info = get_info_pred_distance(coords, y_pred, protein_nodes_xyz, compound_pair_dis_constraint, 
                                  LAS_distance_constraint_mask=LAS_distance_constraint_mask,
                                  n_repeat=1, show_progress=False)
toFile = 'test.sdf'
new_coords = info.sort_values("loss")['coords'].iloc[0].astype(np.double)
write_with_new_coords(mol, new_coords, toFile)
`
